packages/next/src/server/base-server.ts

```diff
@@ -1207,6 +1207,30 @@ export default abstract class Server<
               params = utils.defaultRouteMatches
             }
 
+            // if an action request is bypassing a prerender and we
+            // don't have the params in the URL since it was prerendered
+            // and matched during handle: 'filesystem' not dynamic route
+            // resolving we need to parse the params from the matched-path
+            if (
+              // we can have a collision with /index and a top-level /[slug]
+              matchedPath !== '/index' &&
+              pageIsDynamic &&
+              !paramsResult.hasValidParams &&
+              !isDynamicRoute(matchedPath)
+            ) {
```

Comment:
should this handling be merged with [this](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/base-server.ts#L1146-L1162)? Conceptually seems very similar

Reply:
Moved this closer to that logic and updated comment to explain similarity/difference between the two